### ISSUE TITLE
Required Validation 2 (Ajax) not working

### ISSUE THREAD

**Issue by chkuo2** [kartik-v/bootstrap-fileinput#1826]
Use "Required Validation 2 (Ajax)" sample 
https://plugins.krajee.com/file-count-validation-demo#required-ajax-2
init had file or remove init file and select file , but always show error message "No valid data available for upload.".

**Comment by kartik-v** [comment]
Can you explain what is your issue? It seems to work as intended. If you click upload without selecting a file it will throw no valid data found for upload.

**Comment by chkuo2** [comment]
like this. [https://imgur.com/xlXJ7qD](url)
It had file, but still show error.

**Comment by kartik-v** [comment]
Yes as I mentioned that is expected behavior for upload - since you need to select some file from client to upload - without selecting a file the plugin expects some valid data to upload (this has nothing to do with the `required` function) -- to achieve your use case you may need a custom submit button where you can call the plugin's upload method only after user has selected a file (which you can trap via the input's change event).
